tempodb/encoding/vparquet/wal_block.go

```diff
@@ -251,6 +252,19 @@ type walBlock struct {
 	// Flushed data
 	flushed     []*walBlockFlush
 	flushedSize int64
+	mtx         sync.Mutex
+}
+
+func (b *walBlock) readFlushes() []*walBlockFlush {
+	b.mtx.Lock()
+	defer b.mtx.Unlock()
+	return b.flushed
```

Comment:
does this need to create a new slice and copy the values over?

Reply:
In this case it is ok.  There is an implicit copy of the slice struct (pointer/len), and because `b.flushed` is only appended to it will remain valid.